index.js

```diff
@@ -15,20 +16,51 @@ const port = options.port;
 const dir = options.dir;
 const invite = options.invite;
 const generateInvite = options.generate;
+const key = options.key;
 
-if (!port || !dir) {
-    console.log("Fatal: port (-p) and dir (-d) flags required.");
+if (!port) {
+    console.log("Fatal: port (-p) flag required.");
     process.exit(1);
 }
+
+if (!dir) {
+    console.log("--dir flag required");
+    process.exit(1);
+}
+
 console.log(`Starting on port ${port} using data directory ${dir} ...`);
 
+if (key) {
+    console.log("Loaded key from key command line argument");
+}
+
 function createSbot() {
 
+    let keys = null;
+
+    if (!key || key.length === 0) {
+        // If no key was passed in from a command line argument, generate one
+        console.log("Generating secret key.")
+        keys = ssbKeys.loadOrCreateSync(dir + '/secret');
+    } else {
+        // Passed in as a Base64 encoded environment variable
+        let buff = Buffer.from(key, 'base64');  
+        let keyText = buff.toString('utf-8');
+
+        // The key file generated has # comments at the start, if these are present we remove them
+        // so that we can successfully run JSON.parse on it
+        var jsonText = keyText
+        .replace(/\s*\#[^\n]*/g, '')
+        .split('\n').filter(v => !!v).join('')
+
+        keys = JSON.parse(jsonText);       
+    }
+
     const config = Config('test123', {
         host: "localhost",
         port,
         path: dir,
-        keys: ssbKeys.loadOrCreateSync(dir + '/secret')
+        keys: keys
     })
 
```

Comment:
It seems it's expecting incoming connections to always come from localhost? That won't necessarily always be true - we might be passing a Kubernetes DNS name or similar - (this is what I've listed as ssb launcher url on backend as a new env var). Can we modify this to take an env var potentially but leave localhost as default? Wdyt @mroth?

Reply:
I believe this is the host address that the server binds to (e.g. in case you have multiple network interfaces, etc). Though looking at the ssb-config documentation it is apparently a deprecated option?: https://github.com/ssbc/ssb-config#configuration